web/server/entrypoint.go

```diff
@@ -359,6 +359,19 @@ func (s *robotServer) serveWeb(ctx context.Context, cfg *config.Config) (err err
 		robotOptions = append(robotOptions, robotimpl.WithRevealSensitiveConfigDiffs())
 	}
 
+	shutdownCallbackOpt := robotimpl.WithShutdownCallback(func() {
+		bufSize := 1 << 20
```

Comment:
factor this out so we aren't copying the code from https://github.com/viamrobotics/rdk/pull/4032/files#diff-780f3d307fb668fed30f86c777b47efbbd6848b5aa3282dc1958bc76b2f07ecbR343-R350